src/loki/worker.cc

```diff
@@ -262,6 +267,21 @@ void loki_worker_t::set_interrupt(const std::function<void()>* interrupt_functio
   reader->SetInterrupt(interrupt);
 }
 
+bool loki_worker_t::check_hierarchy_distance(
+    const google::protobuf::RepeatedPtrField<valhalla::Location>& locations) {
+  // Check every location pair
+  for (auto source = locations.begin(); source != locations.end() - 1; ++source) {
+    for (auto target = source + 1; target != locations.end(); ++target) {
+      // Check if arc distance exceeds max distance limit for disabled hierarchy pruning
+      auto arc_distance = to_ll(*source).Distance(to_ll(*target));
+      if (arc_distance > max_distance_disable_hierarchy_culling) {
+        return false;
```

Comment:
I'd prefer to check the distance against the entire route distance, not only every pair. If someone does 20 locations and each 200 kilometers that'll be crazy computation

Reply:
Though not sure how other distance checks handle this, e.g. bike's max_distance_limit, which is pretty much the same as this one.

Reply:
Sure. I change to check the total distance of every pair of locations. For matrix computation, check every pair of source and target (As shown in the new commits). Three actions (matrix_action, route_action, and trace_route_action) call this function.